enterprise/models.py

```diff
@@ -321,6 +368,147 @@ def __repr__(self):
         return self.__str__()
 
 
+class UserDataSharingConsentAudit(TimeStampedModel):
+    """
+    Object that exists to store the canonical state of whether a particular
+    user has given consent for their course data to be shared with a particular
+    single-sign-on identity provider
+    """
+    class Meta(object):
+        app_label = 'enterprise'
+        verbose_name = "Data Sharing Consent Audit State"
+        verbose_name_plural = "Data Sharing Consent Audit States"
+        unique_together = ('user', 'enterprise_customer',)
+
+    NOT_SET = 'not_set'
+    ENABLED = 'enabled'
+    DISABLED = 'disabled'
+    STATE_CHOICES = (
+        (NOT_SET, 'Not set'),
+        (ENABLED, 'Enabled'),
+        (DISABLED, 'Disabled'),
+    )
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+    )
+
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        on_delete=models.CASCADE,
+    )
```

Comment:
I agree that an FK to EnterpriseCustomerUser makes the most sense, and is where we'll go _eventually_. However, that essentially also expands out this story to include decision-making about when to create an EnterpriseCustomerUser. @griffresch, @mattdrayer, thoughts on that?